libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -897,6 +899,71 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
+{
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				if gt(newLen, <maxArrayLength>) {
+					<panic>()
+				}
+
+				let oldLen := <fetchLength>(array)
+
+				if gt(newLen, oldLen) {
+					sstore(array, or(shl(newLen, 1), iszero(lt(newLen, 32))))
+				}
+
+				// Size was reduced, clear end of array
+				if lt(newLen, oldLen) {
```

Comment:
I just realized that we use a different coding style for yul than for c++...

Reply:
That is true. At the beginning I was following what was there, and didn't ask if it was intentional